pkg/version/version.go

```diff
@@ -16,27 +16,67 @@ package version
 
 import (
 	"encoding/json"
+	"fmt"
 	"io"
 )
 
-// A PluginVersioner can encode information about its version
-type PluginVersioner interface {
+// Current reports the version of the CNI spec implemented by this library
+func Current() string {
+	return "0.3.0"
+}
+
+// PluginInfo reports information about CNI versioning
+type PluginInfo interface {
+	// SupportedVersions returns one or more CNI spec versions that the plugin
+	// supports.  If input is provided in one of these versions, then the plugin
+	// promises to use the same CNI version in its response
+	SupportedVersions() []string
+
+	// Encode writes this CNI version information as JSON to the given Writer
 	Encode(io.Writer) error
 }
 
-// BasicVersioner is a PluginVersioner which reports a single cniVersion string
-type BasicVersioner struct {
-	CNIVersion string `json:"cniVersion"`
+type simple struct {
+	CNIVersion_        string   `json:"cniVersion"`
+	SupportedVersions_ []string `json:"supportedVersions,omitempty"`
```

Comment:
Not a conversion, AFAIK, just a hack so that we can have both public fields and functions with the same names.  The fields need to be public so that they get properly marshaled into JSON.  The function names need to match the public interface.

I'm very open to alternative naming schemes.